### ISSUE TITLE
BLD/TST: building fails on `windows-latest`

### ISSUE THREAD

**Issue by neutrinoceros** [yt-project/yt#4972]
Most recent CI jobs show that building yt is currently broken on Windows (at least in CI)
[example logs](https://github.com/yt-project/yt/actions/runs/10612206450/job/29413381013)

here's the relevant excerpt:
```
While building 'yt.utilities.lib.bounding_volume_hierarchy' following error was raised:
   command 'C:\\Program Files\\Microsoft Visual Studio\\2022\\Enterprise\\VC\\Tools\\MSVC\\14.41.34120\\bin\\HostX86\\x64\\cl.exe' failed with exit code 2
  error: command 'C:\\Program Files\\Microsoft Visual Studio\\2022\\Enterprise\\VC\\Tools\\MSVC\\14.41.34120\\bin\\HostX86\\x64\\cl.exe' failed with exit code 2
  [end of output]
```

The specific file that fails to compile hasn't been touched in ~a year so the breaking change must be somewhere else (and most likely external).

**Comment by neutrinoceros** [comment]
Interestingly, #4973 shows that within the context of cibuildwheel, nothing is broken.

**Comment by neutrinoceros** [comment]
I initially thought the key difference was that normal CI installs in editable mode, and that setuptools 74.0.0 (2024-08-27) apparently broke editable mode on windows for us. This hypothesis was however disproven in an experiment I did on my fork, pinning setuptools<74

Now my main suspect is the runner image itself (`windows-latest`, currently aliased to `windows-2022`), which was updated last Monday:
https://github.com/actions/runner-images/releases/tag/win22%2F20240825.1

Indeed, pinning to `windows-2019` fixes CI, though it could only be a temporary patch.
The next step is to report upstream (at https://github.com/actions/runner-images/issues) or identify which existing issue might be relevant.
